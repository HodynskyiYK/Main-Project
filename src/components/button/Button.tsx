import React, { FC } from 'react'
import classnames from 'classnames'
import { IButton } from './buttonTypes'
import styles from './Button.module.scss'

export const Button: FC<IButton> = ({buttonClassName, buttonAction, buttonText, buttonDisabled, buttonType}) => {

    return (
        <button
            className={classnames(styles.btn, `${styles[buttonClassName]}`, {
                [styles.buttonDisabled]: buttonDisabled
            })}
            onClick={buttonAction}
            disabled={buttonDisabled}
            type={buttonType}
        >
            {buttonText}
        </button>
    )
}
