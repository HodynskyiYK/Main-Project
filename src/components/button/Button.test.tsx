import React from 'react'
import { render } from '@testing-library/react'
import { Button } from './Button'

describe('Button tests', () => {

    test('To match snapshot', () => {
        const {asFragment} = render(<Button buttonClassName={'btn'} buttonText={'button'}/>)
        expect(asFragment()).toMatchSnapshot()
    })
})