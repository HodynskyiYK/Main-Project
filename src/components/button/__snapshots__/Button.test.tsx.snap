// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button tests To match snapshot 1`] = `
<DocumentFragment>
  <button
    class="btn btn"
  >
    button
  </button>
</DocumentFragment>
`;
